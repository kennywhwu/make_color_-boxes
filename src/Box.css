.Box {
  display: inline-block;
  border: 1px dotted gray;
}
