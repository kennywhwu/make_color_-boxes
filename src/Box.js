import React, { Component } from 'react';
import './Box.css';

class Box extends Component {
  render() {
    return (
      <div>
        <div
          className="Box"
          style={{
            backgroundColor: this.props.color,
            width: this.props.width,
            height: this.props.height
          }}
        />
        <button onClick={this.props.handleClear}>X</button>
      </div>
    );
  }
}

export default Box;
