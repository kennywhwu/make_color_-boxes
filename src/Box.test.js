import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Box from './Box';

// Smoke test
it('renders without crashing', function() {
  shallow(<Box />);
});

// Snapshot Test
it('matches snapshot', function() {
  let wrapper = shallow(<Box />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
