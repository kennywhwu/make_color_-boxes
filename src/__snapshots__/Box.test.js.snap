// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="Box"
    style={
      Object {
        "backgroundColor": undefined,
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <button>
    X
  </button>
</div>
`;
